src/storage/store/engine_tests.rs

```diff
@@ -2647,4 +2759,203 @@ mod tests {
+        if let Some(proto::hub_event::Body::BlockConfirmedBody(body)) = &first_event.body {
+            assert_eq!(
+                body.block_number,
+                _chunk.header.as_ref().unwrap().height.unwrap().block_number
+            );
+            assert_eq!(
+                body.shard_index,
+                _chunk.header.as_ref().unwrap().height.unwrap().shard_index
+            );
+            assert_eq!(body.timestamp, _chunk.header.as_ref().unwrap().timestamp);
+            assert_eq!(body.total_events, 2); // BLOCK_CONFIRMED + MergeMessage
+        } else {
+            panic!("Expected BlockConfirmedBody");
+        }
+    }
+
+    #[tokio::test]
+    async fn test_block_confirmed_event_sequence_number() {
+        let (mut engine, _tmpdir) = test_helper::new_engine();
+
+        // Register user
+        test_helper::register_user(
+            FID_FOR_TEST,
+            test_helper::default_signer(),
+            test_helper::default_custody_address(),
+            &mut engine,
+        )
+        .await;
+
+        // Create and commit a message
+        let message = messages_factory::casts::create_cast_add(FID_FOR_TEST, "test", None, None);
+        let state_change = engine.propose_state_change(
+            1,
+            vec![MempoolMessage::UserMessage(message.clone())],
+            None,
+        );
+        let _chunk = test_helper::validate_and_commit_state_change(&mut engine, &state_change);
+
+        // Get events from database for this specific block
+        let block_number = _chunk.header.as_ref().unwrap().height.unwrap().block_number;
+        let events = HubEvent::get_events(engine.db.clone(), 0, None, None).unwrap();
+
+        // Find BLOCK_CONFIRMED event for this block
+        let block_confirmed_event = events
+            .events
+            .iter()
+            .find(|e| {
+                e.r#type == proto::HubEventType::BlockConfirmed as i32
+                    && HubEventIdGenerator::extract_height_and_seq(e.id).0 == block_number
+            })
+            .expect("BLOCK_CONFIRMED event not found");
+
+        // Verify BLOCK_CONFIRMED has sequence 0
+        let (event_block_number, sequence) =
+            HubEventIdGenerator::extract_height_and_seq(block_confirmed_event.id);
+        assert_eq!(event_block_number, block_number);
+        assert_eq!(sequence, 0);
+
+        // Verify message event has sequence 1
+        let message_event = events
+            .events
+            .iter()
+            .find(|e| {
+                e.r#type == proto::HubEventType::MergeMessage as i32
+                    && HubEventIdGenerator::extract_height_and_seq(e.id).0 == block_number
+            })
+            .expect("MergeMessage event not found");
+        let (_, sequence) = HubEventIdGenerator::extract_height_and_seq(message_event.id);
+        assert_eq!(sequence, 1);
+    }
+
+    #[tokio::test]
+    async fn test_block_confirmed_event_with_multiple_messages() {
```

Comment:
Can you combine this test with the first test `test_block_confirmed_event_is_always_first`. They're testing the same thing-- you can just have 1 test with multiple messages.